src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/SubscribeToCertificateUpdatesTest.java

```diff
@@ -0,0 +1,212 @@
+import software.amazon.awssdk.aws.greengrass.model.SubscribeToCertificateUpdatesRequest;
+import software.amazon.awssdk.eventstreamrpc.EventStreamRPCConnection;
+import software.amazon.awssdk.eventstreamrpc.StreamResponseHandler;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static com.aws.greengrass.testcommons.testutilities.TestUtils.asyncAssertOnConsumer;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.Mockito.lenient;
+
+@ExtendWith({GGExtension.class, UniqueRootPathExtension.class, MockitoExtension.class})
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class SubscribeToCertificateUpdatesTest extends BaseITCase {
+    private static GlobalStateChangeListener listener;
+    @TempDir
+    Path rootDir;
+    private Kernel kernel;
+    @Mock
+    private GreengrassServiceClientFactory clientFactory;
+    @Mock
+    private GreengrassV2DataClient client;
+
+    public static void subscribeToCertUpdates(GreengrassCoreIPCClient ipcClient,
+                                              SubscribeToCertificateUpdatesRequest request,
+                                              Consumer<CertificateUpdate> consumer) throws Exception {
+        ipcClient.subscribeToCertificateUpdates(request,
+                Optional.of(new StreamResponseHandler<CertificateUpdateEvent>() {
+                    @Override
+                    public void onStreamEvent(CertificateUpdateEvent certificateUpdateEvent) {
+                        consumer.accept(certificateUpdateEvent.getCertificateUpdate());
+                    }
+
+                    @Override
+                    public boolean onStreamError(Throwable error) {
+                        return false;
+                    }
+
+                    @Override
+                    public void onStreamClosed() {
+                    }
+                })).getResponse().get(10, TimeUnit.SECONDS);
+    }
+
+    private static SubscribeToCertificateUpdatesRequest getSampleSubsRequest() {
+        SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest =
+                new SubscribeToCertificateUpdatesRequest();
+        subscribeToCertificateUpdatesRequest.setCertificateOptions(new CertificateOptions().withCertificateType(
+                CertificateType.SERVER));
+        return subscribeToCertificateUpdatesRequest;
+    }
+
+    @BeforeEach
+    void setup() {
+        kernel = new Kernel();
```

Comment:
[not blocking] starting a new kernel on every test is expensive. Try reusing the same instance for all tests and simply updating the configuration if needed.